src/OpenLoco/src/World/Station.cpp

```diff
@@ -1055,6 +1055,44 @@ namespace OpenLoco
         }
     }
 
+#pragma warning(push)
+#pragma warning(disable : 4858)
+    // 0x0048F482
+    void removeTileFromStation(const StationId stationId, const World::Pos3& pos, uint8_t rotation)
+    {
+        auto* station = StationManager::get(stationId);
+        auto findPos = pos;
+        findPos.z |= rotation;
+
+        // Bug mitigation: ensure stationTileSize does not exceed the actual array length
+        station->stationTileSize = std::clamp<uint16_t>(station->stationTileSize, 0, static_cast<uint16_t>(std::size(station->stationTiles)));
+
+        // Find tile to remove
+        auto foundTilePos = std::find(std::begin(station->stationTiles), std::end(station->stationTiles), findPos);
+        if (foundTilePos == std::end(station->stationTiles))
+            return;
```

Comment:
If it is not found we should decrease the tileSize anyway and that will mitigate the bug. Then we catch the situation where you have had 81 tiles deleted some which were found then delete one that was not found.